src/pages/Auth/SignUp.tsx

```diff
@@ -1,112 +1,86 @@
-import { ChangeEvent, useState } from "react";
+import { useEffect, useState } from "react";
+import StepWizard from "react-step-wizard";
 
-import { signUp } from "../../api";
+import { UserData, UserPlan, Workspace } from "./Wizard";
+
+import transitionsStyles from "./Wizard/transitions.module.scss";
+import { useRegisterMutation } from "@api";
+import { AUTH_TOKEN, REFRESH_TOKEN, useLocalStorage } from "@hooks";
 
 export const SignUp = () => {
-  const [userForm, setUserForm] = useState({
-    name: "",
-    email: "",
-    password: "",
-    role: "",
-  });
-  const [workspaceData, setWorkspaceData] = useState({
-    name: "",
-    description: "",
+  const [state, setState] = useState({
```

Comment:
Would  be  nice  to split  one  state to  3  different ( user, workspace, plan )